cartographer/sensor/dispatchable.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2016 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_MAPPING_DISPATCHABLE_H_
+#define CARTOGRAPHER_MAPPING_DISPATCHABLE_H_
+
+#include "cartographer/mapping/trajectory_builder_interface.h"
+#include "cartographer/sensor/data.h"
+
+namespace cartographer {
+namespace sensor {
+
+template <typename DataType>
+class Dispatchable : public Data {
```

Comment:
Next time, please send two separate PRs, one for the refactor.

Reply:
Not sure what you mean by 'refactoring'? I am just moving existing code unchanged into a separate file. That'd make a pretty boring PR ;)